### PR TITLE
Remove external usages of ClassNames (and move it to internal)

### LINKED ISSUE #6583: CodeSpanNameExtractor doesn't produce low-cardinality span names for lambdas
For lambda classes, it generates span names like

`MyClass$$Lambda$772/0x00000008012db3f8.run`

which generally changes with each JVM restart

### PR DESCRIPTION
**Pull request by trask**
Also resolves #6583

### CODE REVIEW

File: instrumentation/spring/spring-webflux-5.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webflux/server/WebfluxSpanNameExtractor.java [hunk 1]
```diff
@@ -6,18 +6,23 @@
 package io.opentelemetry.javaagent.instrumentation.spring.webflux.server;
 
 import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.code.CodeSpanNameExtractor;
 import io.opentelemetry.instrumentation.api.util.SpanNames;
 import org.springframework.web.method.HandlerMethod;
 
 public class WebfluxSpanNameExtractor implements SpanNameExtractor<Object> {
+
+  private final SpanNameExtractor<Object> handlerCodeAttributesGetter =
```
Comment (mateuszrzeszutek) [inline instrumentation/spring/spring-webflux-5.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webflux/server/WebfluxSpanNameExtractor.java:15]:
```suggestion
  private final SpanNameExtractor<Object> handlerSpanNameExtractor =
```

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/code/CodeSpanNameExtractor.java [hunk 2]
```diff
@@ -33,11 +32,15 @@ private CodeSpanNameExtractor(CodeAttributesGetter<REQUEST> getter) {
   public String extract(REQUEST request) {
     Class<?> cls = getter.codeClass(request);
     String className = cls != null ? ClassNames.simpleName(cls) : "<unknown>";
-    String methodName = defaultString(getter.methodName(request));
+    int lambdaIdx = className.indexOf("$$Lambda$");
+    if (lambdaIdx > -1) {
+      // need to produce low-cardinality name, since lambda class names change with each restart
+      className = className.substring(0, lambdaIdx + "$$Lambda$".length());
+    }
```
Comment (mateuszrzeszutek) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/code/CodeSpanNameExtractor.java:39]: Can you add a test case for this?

**Review by laurit — APPROVED** [review]
Lambdas aren't the only ones with the same issue, see https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/3813 Maybe we should truncate after `$$` as it is commonly used in generated class names? Though probably should not do this for `$$anon$` which is used in some scala classes. JDK proxies don't use `$$` and would need to be handled separately.

**Comment by trask** [comment]
> Lambdas aren't the only ones with the same issue, see #3813 Maybe we should truncate after `$$` as it is commonly used in generated class names? Though probably should not do this for `$$anon$` which is used in some scala classes. JDK proxies don't use `$$` and would need to be handled separately.

opened #6595 to track this

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**